### ISSUE TITLE
Protocol information missing when forming new URI

### ISSUE THREAD

**Issue by po5i** [twilio/twilio-ruby#390]
https://github.com/twilio/twilio-ruby/blob/c5d73c29f9aa51ff09395fa419e8657bbc416f6b/lib/twilio-ruby/http/http_client.rb#L28

When I supply proxy information on the constructor, here it makes an invalid URL.  To make it work I have to do this on my custom client:

```ruby
my_request_client = Twilio::HTTP::Client.new(proxy_address, proxy_port, 'http://' + proxy_username, proxy_password)
```

Also is worth to mention that I got a warning: `Warning: use of proxy(new_value) to set connection proxy have been DEPRECATED and will be removed in Faraday 1.0`

**Comment by philnash** [comment]
Hey, sorry for the slow reply. I'm looking into getting this fixed. As you are working with a proxy (which I don't, so I'm not sure what I'm doing here) do you have any suggestions for how to improve the `Twilio::HTTP::Client` and fix this?

Thanks!

**Comment by po5i** [comment]
Hi there @philnash. I'm not an expert in proxy (nor ruby) while I was writing [this tutorial](https://github.com/TwilioDevEd/twilio-ruby-custom-client). But I think a quick fix would be this PR #399
